### ISSUE TITLE
v7_val_t optimization

### ISSUE THREAD

**Issue by eco747** [cesanta/v7#539]
first thanks for your work, it's great.
I used V7 & play with it for a while and here is my remark:
In v7_val_t, you use a smart encoding allowing you to store values: string, double... and you have done some interesting optimizations on that. you have done an optimization for strings <4, char size, = 5 chars size etc. , but there is no optimization on integers - at all. 

1. I write software for a long time and most of the work i do (> 80%) on numbers are integers (i can even say that i never go ahead 32 bit numbers - loops, arrays indexes.
2. my board do not have a fpu, so every time i use a number (an integer one i mean) there is a slow conversion (both senses).
3. C handlers spend there time to convert simple values

My suggestion is : replace V7_TAG_STRING_5 by V7_TAG_INTEGER 
> if the number is (really) of integer type & fit on a 48 bits value, use this optimization.

I will try to do that work on my fork. sadly you do not share your private uncompressed src dir - so i cannot share any suggestion or modification with you.

**Comment by mmikulicic** [comment]
yeah direct support for 32-bit integers something we are considering. No ETA but I'd like to have it. We'd first like to focus on some internal refactoring and stability before adding complexity that will slow down refactoring, so I'd ask you to use some patience for the moment.

We still have room for tag bits so there is no need to replace V7_TAG_STRING_5

-----

w.r.t the unamalgamated code: feel free to make a PR against `v7.c`. Our amalgamation script is reversible and we can incorporate the PR into our code base.

We could also publish the unamalgamated sources, but we had some discussions internally and for now we preferred to have repository with a small number of files.

I understand that might be awkward for those who want to contribute; perhaps we can just include our 'unamalgamation' script so you can unpack the file, edit files with ease and reamagamate them again.

**Comment by eco747** [comment]
amalgamation is a curious idea, but i understand the concept.
I have a remark on it: why did you include v7.h in the amalgamation ?

If i want to create my own objects, i need some 'V7_PRIVATE' functions (ex: v7_set_property - in fact a lot of V7_PRIVATE functions should be useful) the only way i have is to include v7.c in my own code after the #define V7_PRIVATE...
